src/zenml/models/base_models.py

```diff
@@ -47,40 +62,80 @@ def __eq__(self, other: Any) -> bool:
         Returns:
             True if the other object is of the same type and has the same UUID.
         """
-        return self.id == other.id if isinstance(other, DomainModel) else False
-
-    id: UUID = Field(default_factory=uuid4, title="The unique resource id.")
-    created: datetime = Field(
-        default_factory=datetime.now,
-        title="Time when this resource was created.",
-    )
-    updated: datetime = Field(
-        default_factory=datetime.now,
-        title="Time when this resource was last updated.",
-    )
+        if isinstance(other, BaseResponseModel):
+            return self.id == other.id
+        else:
+            return False
 
 
-class UserOwnedDomainModel(DomainModel):
+class UserScopedResponseModel(BaseResponseModel):
     """Base user-owned domain model.
 
     Used as a base class for all domain models that are "owned" by a user.
     """
 
-    user: UUID = Field(
-        title="The id of the user that created this resource.",
+    user: Optional["UserResponseModel"] = Field(
```

Comment:
If it's a user-scoped model, why is this field optional ?

Reply:
I believe the idea here was that a user might be deleted upon which you would not want to delete all of the resources he ever created. However, I think we haven't done any testing yet to see what would actually happen in that case.